File: app-miyou/src/main/java/com/miyou/app/domain/monitoring/port/UsageAnalyticsRepository.java
package com.miyou.app.domain.monitoring.port;

import java.time.Instant;

import com.miyou.app.domain.monitoring.model.UsageAnalytics;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface UsageAnalyticsRepository {
	Mono<UsageAnalytics> save(UsageAnalytics analytics);

	Mono<UsageAnalytics> findById(String pipelineId);

	Flux<UsageAnalytics> findByTimeRange(Instant startTime, Instant endTime);

	Flux<UsageAnalytics> findByModel(String model, int limit);

	Flux<UsageAnalytics> findHighTokenUsage(int tokenThreshold, int limit);

	Flux<UsageAnalytics> findRecent(int limit);

	Mono<Long> countByTimeRange(Instant startTime, Instant endTime);

	Mono<Long> sumTokensByTimeRange(Instant startTime, Instant endTime);

	Mono<Long> count();

	Mono<Long> sumTokens();

	Mono<Double> averageResponseTime();
}

File: domain/src/main/kotlin/com/miyou/app/domain/monitoring/port/UsageAnalyticsRepository.kt
package com.miyou.app.domain.monitoring.port

import com.miyou.app.domain.monitoring.model.UsageAnalytics
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import java.time.Instant

interface UsageAnalyticsRepository {
    fun save(analytics: UsageAnalytics): Mono<UsageAnalytics>

    fun findById(pipelineId: String): Mono<UsageAnalytics>

    fun findByTimeRange(
        startTime: Instant,
        endTime: Instant,
    ): Flux<UsageAnalytics>

    fun findByModel(
        model: String,
        limit: Int,
    ): Flux<UsageAnalytics>

    fun findHighTokenUsage(
        tokenThreshold: Int,
        limit: Int,
    ): Flux<UsageAnalytics>

    fun findRecent(limit: Int): Flux<UsageAnalytics>

    fun countByTimeRange(
        startTime: Instant,
        endTime: Instant,
    ): Mono<Long>

    fun sumTokensByTimeRange(
        startTime: Instant,
        endTime: Instant,
    ): Mono<Long>

    fun count(): Mono<Long>

    fun sumTokens(): Mono<Long>

    fun averageResponseTime(): Mono<Double>
}
